### DISCUSSION TITLE
organize environments using groups --group|-g

### DISCUSSION THREAD

**Discussion by AllanOricil** [forcedotcom/cli#2444] [Ideas]
The inspiration came from Organizer and kubectl

The group identifies the company
<img width="650" alt="image" src="https://github.com/forcedotcom/cli/assets/55927613/938250a5-2276-4568-8504-7c9caa39a0cb">

<img width="347" alt="image" src="https://github.com/forcedotcom/cli/assets/55927613/07617c0e-a816-4c84-9ca1-edd4fae2631c">

````
sf auth:web:login -a foo -g bar => add environment to group
sf org:list -g bar => display all environments from group
sf auth:logout -g foo => logout from all environments in a group
sf org:delete -g foo => delete all environments in a group
````

It is like the `namespace` flag in kubectl.

A new env var called `SF_DEFAULT_GROUP` could be created in a way that `-g|--group` isn't required in every command.

Use cases:

1. I can use the same aliases for the main environments (dev, qa, prod)

This would be possible
group: customer_1
enviroments: dev, qa, prod (aliases)

group: customer_2
environments: dev, qa, prod (aliases)

Instead of
customer_1_dev
customer_2_dev

2. running `sfdx force:org:display` will no longer be necessary to determine the customer the environment belongs to
3. group scratch orgs from multiple devhubs
4. logout or delete all environments under the same group

[a comment by an automated account is not shown]

**Comment by AllanOricil** [comment]
@cristiand391 @WillieRuemmele @mshanemc can u turn this into a discussion

**Comment by AllanOricil** [comment]


**Reply by cristiand391** [reply]
If someone is interested in something like this please comment about your use-cases, current workarounds and/or limitations.

**Reply by AllanOricil** [reply]
I got distracted again 😅

**Comment by AllanOricil** [comment]
I still think this Is useful. It would enable consultants that usually work across many orgs to locate the right environment. Vscode team could then use this feature to implement a new status bar button in vscode to show/select the "group name", for example

Acme Corp | QA

Developers could also use this feature to create custom groups for different features/products they are working on.

@cristiand391 @clairebianchi @mshanemc
